Find and delete keychains with -db extension

mcfedr opened the pull request:
<!-- Thanks for contributing to _fastlane_! Before you submit your pull request, please make sure to check the following boxes by putting an x in the [ ] -->

### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Description
When deleting keychains, search for files with `-db` at the end of the file name.

### Motivation and Context
Since macOS Sierra keychain creates files with an extra -db extension, which delete_keychain action would fail to find.

mcfedr committed:
Find and delete keychains with -db extension

Since macOS Sierra keychain creates files with an extra -db extension

KrauseFx commented:
Thanks for the pull request @mcfedr 👍 There have been quite some changes around the keychain in the past, I'd like @milch's approval on this before merging.

milch commented on fastlane/lib/fastlane/actions/delete_keychain.rb:
```diff
@@ -7,9 +7,16 @@ def self.run(params)
-        search_paths << File.expand_path(File.join("~", "Library", "Keychains", params[:name])) if params[:name]
-        search_paths << File.expand_path(params[:keychain_path]) if params[:keychain_path]
+        if params[:name]
+          search_paths << File.expand_path(params[:name])
```
We recently introduced a new [helper](https://github.com/fastlane/fastlane/blob/5eed4da1559136ff4c490029ebf0b794c826ded2/fastlane_core/lib/fastlane_core/helper.rb#L180) that handles all of this. It searches all the possible locatiosn that you are adding to the search_paths (and more) so you should be able to use that helper instead of this code.

milch reviewed (dismissed):
Hey @mcfedr, thanks for your PR! 

We've moved all keychain searching code to a central location, so please use that method instead of manually building the search list. It seems that we forgot `delete_keychains` when we did that refactoring, so thanks for taking care of it!

mcfedr committed:
Use the keychain path helper in action 'import_certificate'

mcfedr commented:
Hi @milch I have changed things around to use the helper method now, that seems to make a lot more sense!

I noticed that the import_certificate action also had its own search, so and changed that to use the helper.

For a moment I thought the helper wasnt working completely correctly, so there are a could of specs added for it, I assume that isnt a problem

mfurtak commented:
This appears to fix #8543, and LGTM. Is this ready to go in your eyes, @milch?

mfurtak reviewed (approved):
Thanks for this fix!
